### PR TITLE
Some update to the validators

### PR DESCRIPTION
**Pull request by frack113**
* Add **kargs to SigmaRuleValidator validate
* Rework Pattern Tag validator like the tlp

### CODE REVIEW

**Review by nasbench — APPROVED** [review]
LGTM

File: sigma/validators/core/tags.py [hunk 1]
```diff
@@ -102,87 +102,63 @@ def validate(self, rule: SigmaRule) -> List[SigmaValidationIssue]:
...
+        if tag.namespace not in self.allowed_namespace:
+            return [InvalidNamespaceTagIssue([self.rule], tag)]
         return []
 
 
 @dataclass
-class InvalidDetectionTagIssue(SigmaValidationIssue):
-    description: ClassVar[str] = "Invalid detection tagging"
+class InvalidPatternTagIssue(SigmaValidationIssue):
+    description: ClassVar[str] = "The tag use an invalid pattern"
```
Comment (joshnck) [inline sigma/validators/core/tags.py:124]: "The tag use an invalid pattern" -> "The tag used an invalid pattern" or "The tag is using an invalid pattern" (depending on context, 2nd one is probably more accurate)

Comment (joshnck) [reply sigma/validators/core/tags.py:124]: Sorry for being _that guy_ who corrects grammar in a PR

**Review by thomaspatzke — APPROVED** [review]
Moved paramterization to validation class constructor and changed paramterized classes into frozen dataclasses.

**Pull request merged**